streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountDemo.java

```diff
@@ -68,9 +70,10 @@ public static void main(String[] args) throws Exception {
                         return new KeyValue<String, String>(value, value);
                     }
                 })
-                .countByKey(Serdes.String(), "Counts");
+                .countByKey("Counts");
 
-        counts.to("streams-wordcount-output", Serdes.String(), Serdes.Long());
+        // need to override value serde to Long type
+        counts.to(null, Serdes.Long(), "streams-wordcount-output");
```

Comment:
Ah, the `null`'s that are haunting us. :-)

My preference -- though this is matter of style preference -- would be in fact:  Instead of passing `null`, just pass `Serdes.String()` explicitly even though, yes, it is already set as the default serde.  Even though it's not required I think the code reads much better and is less confusing.  For production code people can do of course whatever they want, but for these Streams demo examples we should rather err on the side of clarity (none of the examples are showcasing specifically how to optimize/tune, so I'd prefer clarity over anything else).

So I'd do:

```java
// need to override value serde to Long type
counts.to(Serdes.String(), Serdes.Long(), "streams-wordcount-output");
```